cluster/centos/deployAddons.sh

```diff
@@ -44,6 +44,32 @@ function deploy_dns {
   echo
 }
 
+function deploy_nodelocaldns {
+  echo "Deploying NodeLocalDNS on Kubernetes"
+  cp "${KUBE_ROOT}/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml.sed" nodelocaldns.yaml
+  sed -i -e "s/\\\$DNS_DOMAIN/${DNS_DOMAIN}/g" nodelocaldns.yaml
+  sed -i -e "s/\\\$DNS_SERVER_IP/${DNS_SERVER_IP}/g" nodelocaldns.yaml
+  sed -i -e "s/\\\$LOCAL_DNS_IP/${LOCAL_DNS_IP}/g" nodelocaldns.yaml
+  if [[ -n "${CLUSTER_IP_RANGE:-}" ]]; then
+    sed -i -e "s/\\\$CLUSTER_IP_RANGE/${CLUSTER_IP_RANGE}/g" nodelocaldns.yaml
+  else
+    sed -i -e "s/\\\$CLUSTER_IP_RANGE/0.0.0.0/g" nodelocaldns.yaml
+  fi
+
+  NODELOCALDNS=`eval "${KUBECTL} get daemonset --namespace=kube-system | grep nodelocaldns | cat"`
```

Comment:
Sorry to be pedantic about the shell script:

```
if ${KUBECTL} get pods -o jsonpath='{.items[*].metadata.name}' | tr ' ' "\n" | grep -q "nodelocaldns"; then
...
```

Reply:
done